bindings/go/src/fdb/tuple/tuple.go

```diff
@@ -148,30 +153,75 @@ func (p *packer) encodeBytes(code byte, b []byte) {
 	p.putByte(0x00)
 }
 
-func (p *packer) encodeInt(i int64) {
+func (p *packer) encodeUint(i uint64) {
 	if i == 0 {
-		p.putByte(0x14)
+		p.putByte(intZeroCode)
 		return
 	}
 
-	var n int
+	n := bisectLeft(i)
 	var scratch [8]byte
 
-	switch {
-	case i > 0:
-		n = bisectLeft(uint64(i))
-		p.putByte(byte(intZeroCode + n))
-		binary.BigEndian.PutUint64(scratch[:], uint64(i))
-	case i < 0:
-		n = bisectLeft(uint64(-i))
-		p.putByte(byte(0x14 - n))
-		offsetEncoded := int64(sizeLimits[n]) + i
-		binary.BigEndian.PutUint64(scratch[:], uint64(offsetEncoded))
+	p.putByte(byte(intZeroCode + n))
+	binary.BigEndian.PutUint64(scratch[:], i)
+
+	p.putBytes(scratch[8-n:])
+}
+
+func (p *packer) encodeInt(i int64) {
+	if i >= 0 {
+		p.encodeUint(uint64(i))
+		return
 	}
 
+	n := bisectLeft(uint64(-i))
+	var scratch [8]byte
+
+	p.putByte(byte(intZeroCode - n))
+	offsetEncoded := int64(sizeLimits[n]) + i
+	binary.BigEndian.PutUint64(scratch[:], uint64(offsetEncoded))
+
 	p.putBytes(scratch[8-n:])
 }
 
+func (p *packer) encodeBigInt(i *big.Int) {
+	length := len(i.Bytes())
+	if length > 0xff {
+		panic(fmt.Sprintf("Integer magnitude is too large (more than 255 bytes)"))
+	}
+
+	if i.Sign() >= 0 {
+		intBytes := i.Bytes()
+		if length > 8 {
+			p.putByte(byte(posIntEnd))
+			p.putByte(byte(len(intBytes)))
+		} else {
+			p.putByte(byte(intZeroCode + length))
+		}
+
+		p.putBytes(intBytes)
+	} else {
+		add := new(big.Int).Lsh(big.NewInt(1), uint(length*8))
+		add.Sub(add, big.NewInt(1))
+		transformed := new(big.Int)
+		transformed.Add(i, add)
+
+		intBytes := transformed.Bytes()
+		if length > 8 {
+			p.putByte(byte(negIntStart))
+			p.putByte(byte(length ^ 0xff))
+		} else {
+			p.putByte(byte(intZeroCode - length))
+		}
+
+		for i := len(intBytes); i < length; i++ {
+			p.putByte(0x00)
+		}
```

Comment:
It might be good to have a comment here explaining why this loop is necessary (i.e., that for negative numbers whose absolute value has leading `0xff` bytes, the length of the number might be shorter, so padding with zeroes is necessary to keep the length the same). It took me a while to convince myself that that was the case.

Reply:
Okay, I added a comment to this effect in #915 (see 3c8347e).